File: CPlusPlus/SRP/Before.cpp
#include <iostream>
#include <string>

class User {
public:
    long Id;
    std::string Username;
    std::string Password;
    std::string Email;

    void AddUser(const std::string& username) {
        // Add user logic
        std::cout << "User " << username << " added." << std::endl;
        LogAction("User added");
    }

    void RemoveUser(const std::string& username) {
        // Remove user logic
        std::cout << "User " << username << " removed." << std::endl;
        LogAction("User removed");
    }

private:
    void LogAction(const std::string& message) {
        std::cout << "Log: " << message << std::endl;
    }
};

File: CSharp/SRP/Before.cs
// class User

public class User
{
    public long Id { get; set; }
    public string Username { get; set; }
    public string Password { get; set; }
    public string Email { get; set; }
    
    public void AddUser(string username) 
    {
        // Add user logic
        Console.WriteLine($"User {username} added.");
        LogAction("User added");
    }
    
    public void RemoveUser(string username) 
    {
        // Remove user logic
        Console.WriteLine($"User {username} removed.");
        LogAction("User removed");
    }
    
    private void LogAction(string message) 
    {
        Console.WriteLine($"Log: {message}");
    }
}